Library/Homebrew/on_system.rb

```diff
@@ -128,21 +128,24 @@ def self.setup_base_os_methods(base)
     end
   end
 
-  sig { params(base: Class).void }
+  sig { params(base: T::Class[T.anything]).void }
   def self.setup_macos_methods(base)
     MacOSVersion::SYMBOLS.each_key do |os_name|
       base.define_method(:"on_#{os_name}") do |or_condition = nil, &block|
-        @on_system_blocks_exist = true
+        @on_system_blocks_exist = T.let(true, T.nilable(TrueClass))
```

Comment:
These ivars probably need to use `T.bind` or an rbi to be correctly linked to the `base` class, but i chose to ignore that for now 🙈